onedal/utils/_array_api.py

```diff
@@ -0,0 +1,87 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Tools to support array_api."""
+
+from collections.abc import Iterable
+
+try:
+    from dpctl.tensor import usm_ndarray
+
+    dpctl_available = True
+except ImportError:
+    dpctl_available = False
+
+try:
+    import dpnp
```

Comment:
These checks should be centralized, i.e. flags taken from _device_offload or something. This as it is now would make a maintenance headache.

Reply:
This is already initiated on PR https://github.com/intel/scikit-learn-intelex/pull/1936
Let's do this refactoring there.